doc-content/enterprise-only/employee-rostering/optashift-ER-building-eclipse-proc.adoc

```diff
@@ -0,0 +1,71 @@
+.. Select the `employee-rostering-shared/target/generated-sources` folder and click *OK*.
+. From the main menu, select *Run* > *External Tools* > *External Tools Configurations...*. Under *Program*, create the following launch configurations:
+.. Open OptaWeb Employee Rostering in Chrome:
+*** Name: `Open OptaWeb Employee Rostering in Chrome`
+*** Location: `/usr/bin/google-chrome`
+*** Working Directory: `${workspace_loc:/employee-rostering}`
+*** Arguments: `--incognito http://localhost:8080/gwtui/gwtui.html`
++
+[NOTE]
+====
+You can change the name, location. and arguments to use another browser instead of Chrome.
+====
++
+.. Kill Code Server:
+*** Name: `Kill Code Server`
+*** Location: `/usr/sbin/fuser`
+*** Working Directory: `${workspace_loc:/employee-rostering}`
+*** Arguments: `fuser -k 9876/tcp`
++
+. From the main menu, select *Run* > *Run Configurations...*. 
+. Under *Program*, create the following launch configurations:
+.. OptaWeb Employee Rostering Build:
+*** Name: `OptaWeb Employee Rostering Build`
+*** Base directory: `${workspace_loc:/employee-rostering}`
+*** Goals: `clean install`
+*** Parameter: `gwt:skipCompilation` Value: `true`
++
+.. OptaWeb Employee Rostering Start Code Server:
+*** Name: `OptaWeb Employee Rostering Start Code Server`
+*** Base directory: `${workspace_loc:/employee-rostering}`
+*** Goals: `gwt:codeserver`
+*** Parameter: `gwt:skipCompilation` Value: `true`
++
+.. OptaWeb Employee Rostering Start Webserver:
+*** Name: `OptaWeb Employee Rostering Start Webserver`
+*** Base directory: `${workspace_loc:/employee-rostering}`
+*** Goals: `wildfly:start wildfly:deploy`
+*** Parameter: `gwt:skipCompilation` Value: `true`
++
+.. OptaWeb Employee Rostering Stop Webserver:
+*** Name: `OptaWeb Employee Rostering Stop Webserver`
+*** Base directory: `${workspace_loc:/employee-rostering}`
+*** Goals: `wildfly:shutdown`
+*** Parameter: `gwt:skipCompilation` Value: `true`
+. Under *Launch Group*, create a launch group named `OptaWeb Employee Rostering Run`. Add the following launches to it:
+** `Program::Kill Code Server` Launch mode: *inherit* Post launch action: *Wait until terminated*
+** `Maven Build::OptaWeb Employee Rostering Stop Webserver` Launch mode: *inherit* Post launch action: *Wait until terminated*
+** `Maven Build::OptaWeb Employee Rostering Build` Launch mode: *inherit* Post launch action: *Wait until terminated*
+** `Maven Build::OptaWeb Employee Rostering Start Webserver` Launch mode: *inherit* Post launch action: none
+** `Maven Build::OptaWeb Employee Rostering Start Code server` Launch mode: *inherit* Post launch action: *Wait for console output (regexp)*: `The code server is ready at`
+** `Program::Open OptaWeb Employee Rostering in Chrome` Launch mode: *inherit* Post launch action: none
+. To build, run, and immediately acccess the application, run the `OptaWeb Employee Rostering Run` launch group. You can then change the application and rerun the launch group to test your changes.
```

Comment:
Note: you do not need to re-run the launch group to test changes if all changes were made to gwtui since gwtui uses codeserver (meaning refreshing the web browser would update the code to the latest version).

Reply:
Put the note about it into eclipse build too now (PR to be updated soon)